### PR TITLE
Upgrade react, react-dom to 18

### PR DESCRIPTION
**Pull request by admmasters**
Update react, react-dom and importantly types as well which were broken for Link to 18
Update formik to 2.4.6
Update react-testing-library
Fix a type issue in Link which was triggering errors in the latest react types

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,7 +51,7 @@
     "core-js": "^3.6.5",
     "deprecated-prop-type": "^1.0.0",
     "downshift": "^5.4.7",
-    "formik": "^2.2.0",
+    "formik": "^2.4.6",
```
Comment (admmasters) [inline package.json:54]: Supports react 18 (with less issues)

File: package.json [hunk 2]
```diff
@@ -63,11 +63,9 @@
   },
   "peerDependencies": {
     "@fortawesome/free-solid-svg-icons": "^5.13.0",
-    "@types/styled-components": ">=5.1.25",
```
Comment (admmasters) [inline package.json:66]: We don't need to specify this or typescript as a peer dependency

File: package.json [hunk 3]
```diff
@@ -85,15 +83,15 @@
     "@semantic-release/release-notes-generator": "^10.0.3",
     "@svgr/webpack": "^6.2.1",
     "@swc/core": "^1.3.21",
-    "@testing-library/jest-dom": "5.14.1",
-    "@testing-library/react": "12.0.0",
-    "@testing-library/react-hooks": "7.0.1",
+    "@testing-library/jest-dom": "6.4.8",
```
Comment (admmasters) [inline package.json:86]: Bumped these to support react 18

File: src/components/atoms/List/Item/Item.tsx [hunk 4]
```diff
@@ -16,8 +16,10 @@ const ItemComponent = styled.li`
 
 const Item = ({ icon, className = '', children, ...rest }: ItemProps) => (
   <ItemComponent className={`${icon ? 'has-icon ' : ''}${className}`} {...rest}>
-    {icon}
-    {children}
+    <>
```
Comment (admmasters) [inline src/components/atoms/List/Item/Item.tsx:19]: This is actually a type error now, you need to wrap in a fragment.

File: src/components/atoms/Trustpilot/Stars/Stars.test.tsx [hunk 5]
```diff
@@ -23,7 +23,7 @@ describe('<TrustpilotStars />', () => {
   });
 
   describe('Ratings', () => {
-    const { container, rerender } = render(<Stars />);
+    const { container, rerender } = render(<Stars />, { legacyRoot: true });
```
Comment (admmasters) [inline src/components/atoms/Trustpilot/Stars/Stars.test.tsx:26]: This is the only test in the latest react testing library that requires this.

File: src/setupTests.ts [hunk 6]
```diff
@@ -1,5 +1,6 @@
 import 'jest-styled-components';
 import 'jest-axe/extend-expect';
 import { configure } from '@testing-library/react';
-import '@testing-library/jest-dom/extend-expect';
+import '@testing-library/jest-dom';
```
Comment (admmasters) [inline src/setupTests.ts:4]: This import has changed

File: package.json (hunk 1 above)
Comment (moshie) [reply package.json:54]: Should Formik be a peer dependency?

File: src/components/atoms/Link/Link.tsx [hunk 7]
```diff
@@ -2,9 +2,9 @@ import React from 'react';
 import styled, { css } from 'styled-components';
 import { colors, typography } from '../../../constants';
 import { useThemeContext } from '../../styles/Theme';
-export interface LinkProps
-  extends React.AnchorHTMLAttributes<HTMLAnchorElement>,
-    React.RefAttributes<HTMLAnchorElement> {
+
+// See: https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/forward_and_create_ref/
+export interface LinkProps extends React.AnchorHTMLAttributes<HTMLAnchorElement> {
```
Comment (moshie) [inline src/components/atoms/Link/Link.tsx:7]: nice this has been driving me wild for ages

**Review by moshie — COMMENTED** [review]
Only one comment other than that LGTM

File: package.json (hunk 1 above)
Comment (OlenaKashuba) [reply package.json:54]: no breaking changes in formik, all good?

File: src/components/atoms/Trustpilot/Stars/Stars.test.tsx (hunk 5 above)
Comment (OlenaKashuba) [reply src/components/atoms/Trustpilot/Stars/Stars.test.tsx:26]: how come its only this test, not others

**Review by OlenaKashuba — APPROVED** [review]

**Review by zopa-charlotte-walker — APPROVED** [review]

File: src/components/atoms/Trustpilot/Stars/Stars.test.tsx (hunk 5 above)
Comment (admmasters) [reply src/components/atoms/Trustpilot/Stars/Stars.test.tsx:26]: I don't know. Maybe it means we're doing something not so hot in the Trustpilot component. Or we've written the tests in a weird way.

**Review by zopa-charlotte-walker — APPROVED** [review]

**Pull request merged**